salt/gis.sls

```diff
@@ -0,0 +1,126 @@
+    - install_recommends: false
+    - pkgs:
+      # just some basic packages, not specific to the carto setup.
+      - git 
+      - unzip
+      - curl
+      - build-essential
+      - software-properties-common
+    - refresh: true
+
+# installing the required GIS-software, plus...
+# create the postgresql database and enabling PostGIS
+gis:
+  pkg.installed:
+    - install_recommends: false
+    - pkgs:
+      # here come the goodies; PostGIS is elementary to a current dev. environment.
+      - postgresql-9.3-postgis-2.1
+      - postgresql-contrib-9.3
+      - proj-bin
+      - libgeos-dev
+  # Let's make it easy for our GIS-in-aid software to access our PostGIS database
+  file.blockreplace:
+    - name: /etc/postgresql/9.3/main/pg_hba.conf
+    - marker_start: "# TYPE  DATABASE"
+    - marker_end: '# "local" is'
+    - content: "local   all             gisuser                                 trust"
+  service:
+    - name: postgresql
+    - running
+    - watch:
+      - file: postgresql.conf.work_mem
+      - file: postgresql.conf.maintenance_work_mem
+      - file: gis
+    - require:
+      - pkg: gis
+      - file: gis
+  postgres_user.present:
+    - name: gisuser
+    - superuser: true
+    - createdb: true
+    - require:
+      - service: gis
+  postgres_database.present:
+    - name: gis
+    - db_user: gisuser
+    - require:
+      - postgres_user: gis
+
+# PostGIS: hstore -- extension
+hstore:
+  postgres_extension.present:
+    - db_user: gisuser
+    - maintenance_db: gis
+    - require:
+        - service: gis
+
+# PostGIS: postgis -- extension
+postgis:
+  postgres_extension.present:
+    - db_user: gisuser
+    - maintenance_db: gis
+    - require:
+      - service: gis
+
+# Conservative tweaks to PostgreSQL conf, based on 
+# "Loading OSM data" located at switch2osm.org.
+# TODO: figure out if it's possible to get rid of the other ID, so I can sleep better at night.
+postgresql.conf.work_mem:
+  file.replace:
+    - name: /etc/postgresql/9.3/main/postgresql.conf
+    - pattern: "#work_mem = 1MB"
+    - repl: "work_mem = 16MB"
+
+postgresql.conf.maintenance_work_mem:
+  file.replace:
+    - name: /etc/postgresql/9.3/main/postgresql.conf
+    - pattern: "#maintenance_work_mem = 16MB"
+    - repl: "maintenance_work_mem = 128MB"
+
+# Configure the system to allow overcommitting
+sysctl.overcommit:
+  file.managed:
+    - name: /etc/sysctl.d/60-overcommit.conf
+    - contents: |
+        # Overcommit settings to allow faster osm2pgsql imports
+        vm.overcommit_memory=1
+  cmd.wait:
+    - name: sysctl -p /etc/sysctl.d/60-overcommit.conf
+    - cwd: /
+    - watch:
+      - file: sysctl.overcommit
+
+# OpenStreetMap related repos and packages
+osm:
+  # add the ppa for osm2pgsql with osmctools
+  pkgrepo.managed:
+    - ppa: kakrueger/openstreetmap
+  # install the packages from above-mentioned repo!
+  pkg.latest:
+    - install_recommends: false
+    - refresh: true
+    - pkgs:
+      - osm2pgsql
+      - osmctools
```

Comment:
@pnorman I hadn't noticed that! I updated the comments surrounding it, but it's the only OSM specific ID in the GIS state. So far I've let it stand alone in the event I'd want to `require` it for another step, such as importing during `vagrant provision`. 

(I have another state for that, but I'm not satisfied with it, as it ends up with the whole provisioning taking a loooong time and there's no user feedback while `osm2pgsql` does its job.)